back-end/service.py

```diff
@@ -1,172 +1,301 @@
+                    lines.append(('    - mv {} "$TMPLOC"').format(input_files))
+
+            elif '#copy-backpush-files' in line:
+                if len(backPushfiles) > 0:
+                    input_files = ''
+                    for filename in backPushfiles.values():
+                        input_files += '"$TMPLOC"/"{}" '.format(filename)
+                    lines.append('    - mv {} "$DRONE_BUILD_NUMBER"/'.format(
+                        input_files))
+
+            elif '#get-annexed-files' in line:
+                if len(annexFiles.values()) == 0:
+                    lines.append("    - git annex sync --content")
+                else:
+                    input_files = ""
+                    for filename in annexFiles.values():
+                        input_files += "'{}' ".format(filename)
+                    lines.append('    - git annex get {}'.format(input_files))
+
+            elif '#add-notifications' in line:
+                addNotifications(lines, notifications)
+
+            else:
+                lines.append(line)
+
+    template.close()
+
+    with open(repoPath + '/.drone.yml', 'w+') as config:
+        config.writelines(lines)
+
+    config.close()
+
+    print("Configuration written at " + repoPath + "/.drone.yml")
+
+
+def getRepos(user, token):
+    res = requests.get(
+        path + "/api/v1/users/{}/repos".format(user),
+        headers={'Authorization': 'token ' + str(token)},
+        ).json()
+
+    return res
+
+
+def getRepoData(user, repo, token):
+
+    response = requests.get(
+        path + "/api/v1/repos/{0}/{1}".format(user, repo),
+        headers={'Authorization': 'token ' + str(token)}
+        ).json()
+
+    return response
+
+
+def clone(repo, author, path):
+    clone_path = os.path.join(path, author, repo['name'])
+    os.makedirs(clone_path, exist_ok=True)
+
+    call(['git', 'clone', '--depth=1', repo['clone_url'], clone_path])
+
+    print("Repo cloned at " + clone_path)
+    return clone_path
 
-	print("Repo cleaned from {}".format(path))
 
 def push(path, commitMessage):
-	call(['git', 'add', '.'], cwd=path)
-	call(['git', 'commit', '-m', commitMessage], cwd=path)
-	call(['git', 'push'], cwd=path)
+    call(['git', 'add', '.'], cwd=path)
+    call(['git', 'commit', '-m', commitMessage], cwd=path)
+    call(['git', 'push'], cwd=path)
 
-	print("Updates pushed from {}".format(path))
+    print("Updates pushed from {}".format(path))
 
-def configure(repoName, workflowFiles, backPushFiles, annexFiles, commitMessage, token, auth):
-	repo = getRepoData(auth, auth.username, repoName, token)
 
-	with tempfile.TemporaryDirectory() as temp_clone_path:
-		clone_path = clone(repo, auth.username, temp_clone_path)
-		designWorkflow(workflowFiles, clone_path)
-		designCIConfig(backPushFiles, annexFiles, clone_path)
-		push(clone_path, commitMessage)
-		clean(clone_path)
\ No newline at end of file
+def clean(path):
+    call(['rm', '-rf', path])
```

Comment:
Best not too rely on shell or OS specific commands when the language has it's own implementation.  See https://docs.python.org/3/library/shutil.html#shutil.rmtree

Reply:
Replaced with `shutil.rmtree`